cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1/types.go

```diff
@@ -56,18 +56,13 @@ type AppRepositorySpec struct {
 
 // AppRepositoryAuth is the auth for an AppRepository resource
 type AppRepositoryAuth struct {
-	Header   *AppRepositoryAuthHeader `json:"header,omitempty"`
-	CustomCA *AppRepositoryCustomCA   `json:"customCA,omitempty"`
+	Header        *AppRepoAuthSecret `json:"header,omitempty"`
+	CustomCA      *AppRepoAuthSecret `json:"customCA,omitempty"`
+	RegistryCreds *AppRepoAuthSecret `json:"registryCreds,omitEmpty"`
```

Comment:
This change will also imply adding the corresponding field in  the API description :sweat_smile::
https://github.com/kubeapps/kubeapps/blob/master/dashboard/public/openapi.yaml#L2104-L2111